### PR TITLE
Render combination of standard and alternative script fields in RecordDataFormatter

### PR DESCRIPTION
**Pull request by ThoWagen**
Adds the option to render standard and alternative script fields together when using the RecordDataFormatter.

### CODE REVIEW

File: module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php [hunk 1]
```diff
@@ -377,6 +377,50 @@ protected function getLink($value, $options)
...
+     *
+     * @return string
+     */
+    protected function renderCombineAlt(
+        $data,
+        array $options
+    ) {
+        // Determine the rendering method to use, and bail out if it's illegal:
+        $method = empty($options['combineAltRenderType'])
+            ? 'renderSimple' : 'render' . $options['combineAltRenderType'];
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php:395]: Is it worth adding a test case to cover combineAltRenderType? I don't think I see one...

File: module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php [hunk 2]
```diff
@@ -369,6 +427,13 @@ public function testFormatting(string $function): void
             'c' => 'c',
             'a' => 'a',
             'b' => 'b',
+            'CombineAlt' => 'Standard Title Alternative Title',
+            'CombineAltPrioritizeAlt' => 'Alternative Title Standard Title',
+            'CombineAltDataMethod' => 'Standard Title Other Alternative Title',
+            'CombineAltNoAltFunction' => 'Book',
+            'CombineAltNoAltValue' => '0',
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php:457]: Not sure I understand why '0' is an appropriate value here -- is it intentional that you mock `getBuildings` and don't provide an expectation with a return value for it? Maybe I'm missing or misunderstanding something...

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @ThoWagen -- this looks good to me, but I had some test-related questions.

Also, is there a possible use case where two parallel arrays of values need to get matched up instead of being displayed as side-by-side blocks? That's obviously more complicated and maybe not worth worrying about, but just curious what happens when one or both of the methods return arrays instead of strings. Is that a possibility we need to worry about?

**Comment by demiankatz** [comment]
(I see that getFullTitle is already returning an array in your test, but it contains only one value...)

File: module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php (hunk 1 above)
Comment (ThoWagen) [reply module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php:395]: Added one for renderType "RecordDriverTemplate"

File: module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php (hunk 2 above)
Comment (ThoWagen) [reply module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php:457]: The mock `getBuildings` already existed. I just added `getBuildingsAltScript` to test the behavior when the altScript method exists but does not return anything.

**Comment by ThoWagen** [comment]
I added a test for a method that returns an array.

It's an interesting idea to match entries of arrays. But we don't have a use case for it and as you said it makes things more complicated. So I'd it can be added in a later PR if necessary.

File: module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php (hunk 2 above)
Comment (demiankatz) [reply module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordDataFormatterTest.php:457]: Ahh, I see, should have read the comments a little higher up the page. Sorry! :-)

**Review by demiankatz — APPROVED** [review]
Thanks, @ThoWagen! I think this is safe to include in release 9.1, since it adds functionality without breaking anything. I have updated the [wiki documentation](https://vufind.org/wiki/development:architecture:record_data_formatter) (but please feel free to edit that if you can improve it), and I will merge this now.

**Pull request merged**